api/v1alpha1/proxmoxmachine_types.go

```diff
@@ -203,6 +203,110 @@ type NetworkSpec struct {
 	// +listType=map
 	// +listMapKey=name
 	AdditionalDevices []AdditionalNetworkDevice `json:"additionalDevices,omitempty"`
+
+	// VirtualNetworkDevices defines virtual network devices (e.g. bridges, vlans ...).
+	VirtualNetworkDevices `json:",inline"`
+}
+
+// InterfaceConfig contains all configurables a network interface can have.
+type InterfaceConfig struct {
```

Comment:
Why do we need, IPV config here 😄

Reply:
Because virtual devices can have IPs, and I'm not re-adding that block for each device type that I add (bridges, l2bonds, vlans ...). In fact I'm going to refactor the interface logic a little bit so all devices can have routes added, and this is a stub for doing so. The best thing we can do is to most directly map netplan and reduce anything

Reply:
Netplan has a section for things you can set on every kind of device (or even outwith of devices):
https://netplan.readthedocs.io/en/stable/netplan-yaml/#properties-for-all-device-types
Everything that goes into that section we'd duplicate per network device kind. Therefore we make one datastructure that will hold all possible configurables. For example: `mtu, dhcp, dhcpv6, nameservers` ...

Doing that requires a bigger refactor on the codebase, so I haven't done this here, but that's what this datastructure is for.